### PR TITLE
kubeadm: fix a bug where the kubelet cert rotation was turned off

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:
A recent [PR](https://github.com/kubernetes/kubernetes/pull/81903) added warnings for KubeletConfiguration and
KubeProxyConfiguration fields that kubeadm cares about and
does not recommend the user modifying them. Kubelet's
"rotateCertificates" cannot be handled using this function
as there is not way to figure out if the user has set it explicitly to
"false". Hardcode the value to "true" and add a comment about that.
Note that "true" was already the value before https://github.com/kubernetes/kubernetes/pull/81903.

Also apply the following changes to warnDefaultComponentConfigValue()
calls:
- use a local "kind" variable that defines the Kind we are warning about.
- fix wrong paths to fields.

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:
kind's CI signal thankfully caught this because kind was watching stderr and klog.Warning prints to stderr:
https://github.com/kubernetes/kubernetes/issues/82097

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/kind bug
/priority critical-urgent
cc @rosti 
cc @BenTheElder @aojea

### CODE REVIEW

File: cmd/kubeadm/app/componentconfigs/defaults.go [hunk 1]
```diff
@@ -120,64 +123,64 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
...
-		warnDefaultComponentConfigValue(externalkubeletcfg.Kind, "read-only-port", kubeletReadOnlyPort, externalkubeletcfg.ReadOnlyPort)
+		warnDefaultComponentConfigValue(kind, "readOnlyPort", kubeletReadOnlyPort, externalkubeletcfg.ReadOnlyPort)
 	}
 
-	if externalkubeletcfg.RotateCertificates != kubeletRotateCertificates {
-		warnDefaultComponentConfigValue(externalkubeletcfg.Kind, "rotate-certificates", kubeletRotateCertificates, externalkubeletcfg.RotateCertificates)
-	}
+	// We cannot show a warning for RotateCertificates==false and we must hardcode it to true.
+	// There is no way to determine if the user has set this or not, given the field is a non-pointer.
+	externalkubeletcfg.RotateCertificates = true
```
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/componentconfigs/defaults.go]: `true` was the default before the PR that added the warnings:
https://github.com/kubernetes/kubernetes/pull/81903/files#diff-5d5570f188cf5eb502aa81ba486fc736L114

[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/componentconfigs/defaults.go]: yes, enforcing a default value of `true`, for a kubelet boolean field for which we cannot determine if the user has set it explicitly to `false` or if the language defaulted a structure field to `false`.

this has to be fixed for kubelet v1beta2.

**Comment by neolit123** [comment]
thanks
/hold cancel

**Review by rosti — APPROVED** [review]
Thanks @neolit123 !
/lgtm

File: cmd/kubeadm/app/componentconfigs/defaults.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (b0b0haha) [reply cmd/kubeadm/app/componentconfigs/defaults.go]: > 
I'm confused that why there is no way to determine if the user has set this field or not.

**Pull request merged**